libs/pilight/protocols/433.92/hideki.c

```diff
@@ -0,0 +1,512 @@
+		wind_factor = my_settings->wind_factor;
+		rain_factor = my_settings->rain;
+
+		if((battery_ok == 0) && (my_settings->ignore_temp)) {
+			ignore_temperature = 1; // workaround for case that battery=0: temperature value might be invalid in that case
+		}
+
+		if(my_settings->calibration > 0) {
+			/* still in calibration phase to memorize correct sensor type */
+#ifdef HIDEKI_DEBUG
+			logprintf(LOG_DEBUG, "HIDEKI parseCode(): calibration: %d, type: %d -> %d",
+				  my_settings->calibration, my_settings->sensortype, sensortype);
+#endif
+			my_settings->calibration--;
+			if(my_settings->sensortype == HIDEKI_UNKNOWN) {
+				my_settings->sensortype = sensortype;
+			} else if(my_settings->sensortype != sensortype) {
+				my_settings->sensortype = sensortype;
+				my_settings->calibration = CALIBRATION_COUNT;
+			}
+		} else if(my_settings->sensortype != sensortype) {
+			/* calibration is finished and detected sensor type does not match */
+			logprintf(LOG_DEBUG, "HIDEKI parseCode(): wrong sensor type found: %d (expected: %d) -> ignoring data",
+				  sensortype, my_settings->sensortype, sensortype);
+			return;	// ignore the data
+		}
+	} else {
+		/* no valid settings/configuration found -> the received data do not belong to a configured device from config.json
+		 * -> usually we could return here, but in that case the tool "pilight_receive" might be unable to receive other data
+		 * than locally configured... */
+	}
+	
+	if(sensortype == HIDEKI_TS04) {
+		temperature = (double)temp/10 + temp_offset;
+		humidity = ((packet[6] & 0xF0) >> 4) * 10 + (packet[6] & 0x0F) + humi_offset;
+
+		hideki->message = json_mkobject();
+		if(ignore_temperature == 0) {
+			json_append_member(hideki->message, "temperature", json_mknumber(temperature, 1));
+		}
+		json_append_member(hideki->message, "humidity", json_mknumber(humidity, 0));
+
+		json_append_member(hideki->message, "battery", json_mknumber(battery_ok, 0));
+		json_append_member(hideki->message, "channel", json_mknumber(channel, 0));
+		json_append_member(hideki->message, "rc", json_mknumber(rc, 0));
+	} else if(sensortype == HIDEKI_WIND) {
+		const uint8_t wd[] = { 0, 15, 13, 14, 9, 10, 12, 11, 1, 2, 4, 3, 8, 7, 5, 6 };
+		temperature = (double)temp/10 + temp_offset;
+		wind_strength = (double)(packet[9] & 0x0F) * 100 + ((packet[8] & 0xF0) >> 4) * 10 + (packet[8] & 0x0F); // unit still unclear
+		wind_strength *= wind_factor; // user defined adjustment
+		// wind strength in km/h = wind_strength*0.160934
+		wind_direction = (double)(wd[((packet[11] & 0xF0) >> 4)] * 225);
+		wind_direction /= 10;
+		wind_direction += wind_dir_offset; // user defined adjustment
+#if 0
+		if(wind_direction > 360.0) {
+			wind_direction -= 360.0;
+		}
+#endif
+		hideki->message = json_mkobject();
+		if(ignore_temperature == 0) {
+			json_append_member(hideki->message, "temperature", json_mknumber(temperature, 1));
+		}
+		json_append_member(hideki->message, "wind strength", json_mknumber(wind_strength, 1));
```

Comment:
The wind options doesn't have spaces.

Reply:
Oh, I thought this argument is considered as human readable string. I can replace the space with dash, ok?